runtest.py

```diff
@@ -232,24 +235,96 @@ def get_test_env(self, **kwargs):
         result['GLOBAL_CONFIG_FP'] = self.global_config_fp
         result['FIREFOX_CONFIG_FP'] = self.firefox_config_fp
         result['CHROME_CONFIG_FP'] = self.chrome_config_fp
-        result['ONLINE_CONFIG_FP'] = self.online_config_fp
+        result['UPLOAD_CONFIG_FP'] = self.upload_config_fp
         result['SUITE_RESULT_DP'] = str(self.suite_result_dp)
         result['FIREFOX_PROFILE_PATH'] = self._firefox_profile_path
         result['CHROME_PROFILE_PATH'] = self._chrome_profile_path
 
-        if self.online_config['perfherder-revision']:
-            result['PERFHERDER_REVISION'] = str(self.online_config['perfherder-revision'])
+        if self.upload_config['perfherder-revision']:
+            result['PERFHERDER_REVISION'] = str(self.upload_config['perfherder-revision'])
```

Comment:
in `get_test_env()` method, we should raise Exception if there is no `perfherder-revision`, `perfherder-pkg-platform`.